### DISCUSSION TITLE
Cannot add to HACS

### DISCUSSION THREAD

**Discussion by ojessie** [OStrama/judo_rest_api#2] [General]
Thank you for this new approach to integrate Judo. Highly appreciated. Unfortunately, I cannot add it to HACS. Whatever I do, for whatever I search nothing is there. Keep my fingers crossed that you check this issue. 
Thanks

**Comment by OStrama** [comment]
Currently this is at the beginning and only has some sensors. Please try adding this link (I hhink the old one in the readme was wrong..

https://github.com/OStrama/judo_rest_api

As custom repository in hacs (3 dots top right)
Then you can search for it in hacs and download the pre-release. At next you can add the integration in HA.

**Comment by ojessie** [comment]
Thanks for the immediate response ... but it's still not possible.
I'm having several additional repositories, adding yours is always the 
same ... it seems to be added but after the restart it's gone.



------ Originalnachricht ------
Von "OStrama" ***@***.***>
An "OStrama/judo_rest_api" ***@***.***>
Cc "ojessie" ***@***.***>; "Author" 
***@***.***>
Datum 17.12.2024 10:33:01
Betreff Re: [OStrama/judo_rest_api] Cannot add to HACS (Discussion #2)

**Reply by OStrama** [reply]
Oh, do you see any errors in the log?

**Reply by OStrama** [reply]
Another point, it is currently only a pre-release. Did you select the pre-release:
![{D0DFBE6B-BE7D-44C1-8BB6-EC45567FA6FF}](https://github.com/user-attachments/assets/352dfe1c-ad6b-4b5c-967c-0efa3ebbb5b2)

**Comment by ojessie** [comment]
Yess, I got it! I got 2 devices and 7 entities. Each of them has an unknow stage. 
I really hope that you are continuing with the development, because then I'll stop trying to get this one work ... https://github.com/www-ShapeLabs-de/Judo-i-soft-save-plus-to-mqtt-bridge?tab=readme-ov-file

**Reply by OStrama** [reply]
Yes, that‘s my intention, too. Especially to have this all local. Yes, I’ll continue until all items mentioned in the documentation are available at least for my device type. However, I do this in my spare time together with the heatpump integration, so some breaks may happen ;-)

**Comment by OStrama** [comment]
Any errors in the log?
I have a isoft SAFE+, maybe the addresses are different for other devices. Could you tell me the type of your device?

**Reply by ojessie** [reply]
I'm having Isoft Safe+ ... the same one ...

checked all logs, no errors regarding your integration ...

**Reply by ojessie** [reply]
Update installed log viewer ... there are errors ... 
2024-12-17 13:14:10.796 WARNING (MainThread) [custom_components.judo_rest_api.restobject] Connection to Judo Water Treatment failed
2024-12-17 13:14:10.796 WARNING (MainThread) [custom_components.judo_rest_api.coordinator] connection to Judo Water treatment failed
2024-12-17 13:14:11.061 WARNING (MainThread) [custom_components.judo_rest_api.restobject] Connection to Judo Water Treatment failed
2024-12-17 13:14:11.061 WARNING (MainThread) [custom_components.judo_rest_api.coordinator] connection to Judo Water treatment failed

**Reply by ojessie** [reply]
another update: .... remarkable is, that I cannot connect with my browser to the connectivity module, using the standard credentials .... any idea how to reset the module?

**Reply by ojessie** [reply]
well, well, well ... all my bad ... you won't believe it. I have an "old" connectivity module without Wlan and obviously without the API interface .... I have to replace it. Sorry

**Reply by OStrama** [reply]
So then we know at least the reason ;-). Unfortunately, I have no idea how to reset the PW, in the description nothing is mentioned: https://judo.eu/app/downloads/files/de/8203521/manuals/1702574_202405.pdf. Maybe, there's a reset button or similar on the module?

**Reply by ojessie** [reply]
Back in race ;-) after replacing the connectivity module all entities showing the corresponding information.

**Reply by OStrama** [reply]
Wow, that was fast! Nicwe to hear that it works now. So you can try out the next pre-release if you want ;-)

**Comment by ojessie** [comment]
Great! I've to confess I'm not a coder or developer ... kind of advanced 
user ;-) if I could also participate on your heatpump integration would 
be marvelous. I'm fighting and searching since days to find a kind of 
"template" or blueprint for my additional heading which is basically a 3 
way valve mixer and a pump. The challenge is the flow temperature is 
based on a heating curve depending on the outside temperature .... will 
see if I find something ... and yes the new version is working 👍👍👋


------ Originalnachricht ------
Von "OStrama" ***@***.***>
An "OStrama/judo_rest_api" ***@***.***>
Cc "ojessie" ***@***.***>; "Author" 
***@***.***>
Datum 17.12.2024 20:38:17
Betreff Re: [OStrama/judo_rest_api] Cannot add to HACS (Discussion #2)

**Reply by OStrama** [reply]
The heatpump integration is especially designed for the modbus interface of Weishaput heatpumps. In this world every brand uses its own way to make their devices smart..

**Reply by ojessie** [reply]
I know, I'm having a KNV heatpump already integrated with modbus. Quite happy with it.
